nixos/modules/services/web-apps/tachidesk-server.nix

```diff
@@ -0,0 +1,227 @@
+      };
+    };
+
+    systemd.tmpfiles.rules = [
+      "d '${cfg.dataDir}/.local/share/Tachidesk' 0700 ${cfg.user} ${cfg.group} - -"
+    ];
+
+    systemd.services.tachidesk-server =
+      let
+        #  HOCON is a JSON superset that tachidesk-server use for configuration
+        toHOCON = attr: (
+          let
+            attrType = builtins.typeOf attr;
+          in
+          if builtins.elem attrType [ "string" "path" "int" "float" ] then ''"${toString attr}"''
+          else if attrType == "bool" then lib.boolToString attr
+          # attrs, list, lambda, null
+          else
+            throw ''
+              [tachidesk-server]: invalid config value type '${attrType}'.
+            ''
+        );
+        configFile = pkgs.writeText "server.conf" (lib.strings.concatStrings [
+          ''
+            server.systemTrayEnabled = "${lib.boolToString cfg.enableSystemTray}"
+
+            server.ip = "${cfg.config.ip}"
+            server.port = "${toString cfg.config.port}"
+
+            server.downloadAsCbz = ${lib.boolToString cfg.config.downloadAsCbz}
+            server.localSourcePath = "${cfg.config.localSourcePath}"
+
+            server.basicAuthEnabled = ${lib.boolToString cfg.config.basicAuth.enable}
+          ''
+
+          (
+            if cfg.config.basicAuth.enable then
+              (if builtins.isNull cfg.config.basicAuth.username || builtins.isNull cfg.config.basicAuth.passwordFile then
+                throw ''
+                  [tachidesk-server]: the username and the password file cannot be null when the basic auth is enable
+                '' else ''
+                server.basicAuthUsername = "${cfg.config.basicAuth.username}"
+                server.basicAuthPassword = "$TACHIDESK_SERVER_BASIC_AUTH_PASSWORD"
+              '') else ""
+          )
+
+          (lib.pipe cfg.extraConfig [
+            (lib.filterAttrs (_: x: x != null))
+            (lib.mapAttrsToList (name: value: ''${name} = ${toHOCON value}''))
+            lib.concatLines
+          ])
+        ]);
+      in
+      {
+        description = "A free and open source manga reader server that runs extensions built for Tachiyomi.";
+
+        wantedBy = [ "multi-user.target" ];
+        wants = [ "network-online.target" ];
+        after = [ "network-online.target" ];
+
+        environment = {
+          TACHIDESK_SERVER_BASIC_AUTH_PASSWORD = if cfg.config.basicAuth.enable then "${lib.fileContents cfg.config.basicAuth.passwordFile}" else "";
+        };
+
+        serviceConfig = {
+          User = cfg.user;
+          Group = cfg.group;
+
+          Type = "simple";
+          Restart = "on-failure";
+          EnvironmentFile = "";
+          ExecStartPre = [
+            "${pkgs.envsubst}/bin/envsubst -i ${configFile} -o ${cfg.dataDir}/.local/share/Tachidesk/server.conf"
+          ];
+          ExecStart = "${cfg.package}/bin/tachidesk-server -Dsuwayomi.tachidesk.config.server.rootDir=${cfg.dataDir}";
```

Comment:
```suggestion
            "${lib.getExe pkgs.envsubst} -i ${configFile} -o ${cfg.dataDir}/.local/share/Tachidesk/server.conf"
          ];
          ExecStart = "${lib.getExe cfg.package} -Dsuwayomi.tachidesk.config.server.rootDir=${cfg.dataDir}";
```

Reply:
I was not aware that this existed, it is changed